address-space-controller/src/main/java/io/enmasse/controller/RouterStatusCache.java

```diff
@@ -0,0 +1,275 @@
+        for (int i = 0; i < routerPods.size(); i++) {
+            try {
+                RouterStatus status = service.take().get();
+                if (status != null) {
+                    routerStatusList.add(status);
+                }
+            } catch (Exception e) {
+                log.info("Error requesting router status. Ignoring", e);
+                eventLogger.log(RouterCheckFailed, e.getMessage(), Warning, ControllerKind.AddressSpace, addressSpace.getMetadata().getName());
+            }
+        }
+        return routerStatusList;
+    }
+
+    private RouterStatus collectStatus(RouterManagement routerManagement, Pod router) {
+        try {
+
+            int port = 0;
+            for (Container container : router.getSpec().getContainers()) {
+                if (container.getName().equals("router")) {
+                    for (ContainerPort containerPort : container.getPorts()) {
+                        if (containerPort.getName().equals("amqps-normal")) {
+                            port = containerPort.getContainerPort();
+                        }
+                    }
+                }
+            }
+
+            if (port != 0) {
+                // Until the connector entity allows querying for the status, we have to list
+                // all connections and match with the connector host.
+                Map<RouterEntity, List<List<?>>> response = routerManagement.query(router.getStatus().getPodIP(), port, entities);
+
+                RouterConnections connections = null;
+                if (response.containsKey(connection)) {
+                    connections = collectConnectionInfo(response.get(connection));
+                }
+
+                List<String> neighbours = null;
+                if (response.containsKey(node)) {
+                    neighbours = filterOnAttribute(String.class, 0, response.get(node));
+                }
+
+                long undeliveredTotal = 0;
+                if (response.containsKey(link)) {
+
+                    List<String> linkTypes = filterOnAttribute(String.class, 0, response.get(link));
+                    List<UnsignedLong> undelivered = filterOnAttribute(UnsignedLong.class, 1, response.get(link));
+                    for (int i = 0; i < linkTypes.size(); i++) {
+                        if ("inter-router".equals(linkTypes.get(i))) {
+                            undeliveredTotal += undelivered.get(i) != null ? undelivered.get(i).longValue() : 0;
+                        }
+                    }
+                }
+                return new RouterStatus(router.getMetadata().getName(), connections, neighbours, undeliveredTotal);
+            }
+        } catch (Exception e) {
+            log.info("Error requesting registered topics from {}. Ignoring", router.getMetadata().getName(), e);
+        }
+        return null;
+    }
+
+    /*
+     * Until the connector entity allows querying for the status, we have to go through all connections and
```

Comment:
Isn't DISPATCH-1419 sufficient?

Reply:
I just copied this from the existing code, but you're right. I will update it to avoid querying all connections.